plugins/ua_pubsub_udp.c

```diff
@@ -530,19 +529,13 @@ UA_PubSubChannelUDPMC_receive(UA_PubSubChannel *channel, UA_ByteString *message,
         }
 
         UA_DateTime now = UA_DateTime_nowMonotonic();
-        if(channelConfigUDPMC->ai_family == PF_INET){
-            ssize_t messageLength = UA_recvfrom(channel->sockfd, (message->data + dataLength), remainingMessageLength, 0, NULL, NULL);
-            if(messageLength > 0){
-                dataLength += (size_t)messageLength;
-                remainingMessageLength -= (size_t)dataLength;
-            } else {
-                retval = UA_STATUSCODE_BADINTERNALERROR;
-                break;
-            }
-#if UA_IPV6
+        ssize_t messageLength = UA_recvfrom(channel->sockfd, (message->data + dataLength), remainingMessageLength, 0, NULL, NULL);
+        if(messageLength > 0){
+            dataLength += (size_t)messageLength;
+            remainingMessageLength -= (size_t)dataLength;
         } else {
-        //TODO implement recieve for IPv6
-#endif
+            retval = UA_STATUSCODE_BADINTERNALERROR;
```

Comment:
Add a UA_LOG_WARNING in case the receive fails.
See above for a copy/paste example.

The same for the select-call above.

Reply:
Make sense. I'll add more logging and a test case. Thanks!